### DISCUSSION TITLE
Input channel layout with layout optimization

### DISCUSSION THREAD

**Discussion by francis2tm** [microsoft/onnxruntime#16924] [General]
Hello,
So, right now, I have a model that receives images which are channel last (NHWC). The ONNX default is NCHW. Therefore, my model needs to have a transpose op in the beginning:
![image](https://github.com/microsoft/onnxruntime/assets/43780565/ca532e21-29d2-46c1-a4d3-60088d17318f)

However, in [the docs](https://onnxruntime.ai/docs/performance/model-optimizations/graph-optimizations.html#layout-optimizations) you say that after the model is optimized, the "graph uses NCHWc layout instead of NCHW layout".
So... The pretty expensive transpose OP added in the beginning just becomes useless, correct?
I'd like very much to remove that transpose OP. Any ideas?

**Comment by carzh** [comment]
Hmm I'm not sure if there is any current functionality to remove an op in an ONNX graph -- you could try some third-party libraries like [ONNX modifier](https://github.com/ZhangGe6/onnx-modifier) or [ONNX tool](https://github.com/ThanatosShinji/onnx-tool)
